lunatrace/bsl/backend/src/github/projects.ts

```diff
@@ -68,48 +138,93 @@ function getGithubAccessToken(identity: Identity) {
   return githubProvider.initial_access_token;
 }
 
-githubApiRouter.post('/github/user/projects', async (req, res) => {
+export interface KratosIdentityConfig {
+  providers: KratosIdentityProvider[];
+}
+
+export interface KratosIdentityProvider {
+  initial_id_token: string;
+  subject: string;
+  provider: string;
+  initial_access_token: string;
+  initial_refresh_token: string;
+}
+
+githubApiRouter.post('/webhook/github/login', async (req, res) => {
   const kratosConfig = new Configuration({
     basePath: 'http://localhost:4434',
   });
 
   const kratosApiClient = V0alpha2ApiFactory(kratosConfig);
 
-  const userId = req.body.user_id;
+  const userId = req.body.ctx.identity.id;
 
   const identity = await kratosApiClient.adminGetIdentity(userId, ['oidc']);
 
   const accessToken = getGithubAccessToken(identity.data);
 
-  const graphqlWithAuth = graphql.defaults({
-    headers: {
-      authorization: `token ${accessToken}`,
-    },
-  });
+  const github = generateGithubGraphqlClient(accessToken);
 
-  try {
-    const response = await graphqlWithAuth(
-      `query ($login: String!) {
-      organization(login: $login) {
-        repositories(privacy: PRIVATE) {
-          totalCount
-        }
-      }
-    }`,
-      { login: 'breadchris' }
-    );
-    res.send(response);
-  } catch (e) {
-    console.error(e);
-    if (e instanceof GraphqlResponseError) {
-      res.send({
-        error: e.message,
-      });
-    } else {
-      // TODO (cthompson) what other error types exist?
-      res.send({
-        error: e,
-      });
-    }
+  let orgsAfter: string | null | undefined = undefined;
+  let allUserOrgs: GetUserOrganizationsQuery | null = null;
+  let moreDataAvailable = true;
+
+  while (moreDataAvailable) {
+    const userOrgs: GetUserOrganizationsQuery = await github.GetUserOrganizations({
+      orgsAfter: orgsAfter,
+    });
+
+    allUserOrgs = deepmerge(allUserOrgs || {}, userOrgs);
+
+    if (!userOrgs.viewer) break;
```

Comment:
Nit: Use braces instead of one liners (Apple once had a bad bug because of this syntax once, so I try to use braces now 100% of the time. We should really just make the linter do this for us.)

Reply:
Good catch I just set this in the linter.  Luckily its always auto fixable